lib/client/keyagent.go

```diff
@@ -199,38 +207,57 @@ func (a *LocalKeyAgent) AddHostSignersToCache(hostSigners []services.CertAuthori
 	return nil
 }
 
+// UserRefusedHosts returns 'true' if a user refuses connecting to remote hosts
+// when prompted during host authorization
+func (a *LocalKeyAgent) UserRefusedHosts() bool {
+	return len(a.noHosts) > 0
+}
+
 // CheckHostSignature checks if the given host key was signed by one of the trusted
 // certificaate authorities (CAs)
-func (a *LocalKeyAgent) CheckHostSignature(hostId string, remote net.Addr, key ssh.PublicKey) error {
+func (a *LocalKeyAgent) CheckHostSignature(host string, remote net.Addr, key ssh.PublicKey) error {
+	hostPromptFunc := func(host string, key ssh.PublicKey) error {
+		userAnswer := "no"
+		if !a.noHosts[host] {
+			fmt.Printf("The authenticity of host '%s' can't be established. "+
+				"Its public key is:\n%s\nAre you sure you want to continue (yes/no)? ",
+				host, ssh.MarshalAuthorizedKey(key))
```

Comment:
I'm curious why you went with `authorized_keys` format here over `known_hosts` format? Also why not a key fingerprint like OpenSSH does?